### DISCUSSION TITLE
TypeError: The "iterable" argument must be an instance of Iterable. Received an instance of ReadableStream

### DISCUSSION THREAD

**Discussion by jasonChen1982** [node-fetch/node-fetch#1585] [Q&A]
**Reproduction**

```js
    const file = fileFromSync(filePath);
    const formData = new FormData();
    formData.append('file', file, path.basename(filePath));
```

1. I use upon code in electron, but get a `TypeError: The "iterable" argument must be an instance of Iterable. Received an instance of ReadableStream`
2. maybe is there wrong ? 
![image](https://user-images.githubusercontent.com/8683772/173488815-4d33d904-3953-42ba-add8-372e55983186.png)
3. can you tell me node-fetch can work in electron (which has browser\node both)

**Expected behavior**

**Screenshots**
![image](https://user-images.githubusercontent.com/8683772/173489097-9c86e6b8-d6f9-4ee4-92bd-8670dc9283b6.png)

**Your Environment**

| software         | version
| ---------------- | -------
| node-fetch       | 3.2.6
| node             |
| npm              |
| Operating System |

**Additional context**

**Comment by jimmywarting** [comment]
If ReadableStream is a native chrome stream, then you probably have to add a polyfill for making them async readable.

```js
ReadableStream.prototype[Symbol.asyncIterator] ??= async function* () {
  const reader = this.getReader()
  try {
    while (true) {
      const {done, value} = await reader.read()
      if (done) return
      yield value
    }
  }
  finally {
    reader.releaseLock()
  }
}
```

**Comment by jasonChen1982** [comment]
yeah, thanks @jimmywarting , I get your mean. right, it is `fetch-blob` had a little bit unrigorous check. I hack it outside and then it work correct.
```js

if (!globalThis.ReadableStream) {
  // `node:stream/web` got introduced in v16.5.0 as experimental
  // and it's preferred over the polyfilled version. So we also
  // suppress the warning that gets emitted by NodeJS for using it.
  try {
    const process = require('node:process')
    const { emitWarning } = process
    try {
      process.emitWarning = () => {}
      Object.assign(globalThis, require('node:stream/web'))
      process.emitWarning = emitWarning
    } catch (error) {
      process.emitWarning = emitWarning
      throw error
    }
  } catch (error) {
    // fallback to polyfill implementation
    Object.assign(globalThis, require('web-streams-polyfill/dist/ponyfill.es2018.js'))
  }
}
```
